src/picongpu/include/plugins/output/GatherSlice.hpp

```diff
@@ -94,7 +104,11 @@ struct GatherSlice
         MPI_CHECK(MPI_Group_free(&group));
         MPI_CHECK(MPI_Group_free(&newgroup));
 
-        return mpiRank == 0;
+        /* avoid that only rank zero is the master
+         * this reduces the load of rank zero
+         */
+        masterRank = (masterRankOffset % numRanks);
```

Comment:
I move `masterRankOffset++` down to this lines because `numRanks` is used for the modulo operation and not known at the beginning of the method.